src/pwmgeneration-sine.cpp

```diff
@@ -187,14 +188,9 @@ s32fp PwmGeneration::LimitCurrent()
 s32fp PwmGeneration::GetIlMax(s32fp il1, s32fp il2)
 {
    s32fp il3 = -il1 - il2;
-   s32fp offset = SineCore::CalcSVPWMOffset(il1, il2, il3) / 2;
-   offset = ABS(offset);
-   il1 = ABS(il1);
-   il2 = ABS(il2);
-   il3 = ABS(il3);
-   s32fp ilMax = MAX(il1, il2);
-   ilMax = MAX(ilMax, il3);
-   ilMax -= offset;
+   s32fp ilMax = FP_MUL(il1, il1) + FP_MUL(il2, il2) + FP_MUL(il3, il3);
```

Comment:
I'm using this code, but it occurs to me that this operation may not be safe. A current exceeding 1024A (32768 FP) may overflow the s32fp during the multiplication or add operations. Any ideas?